37974 - add key components to the modules introduction (#217)
valtimo-platform/valtimo-documentation: introduction/modules/modules.md

framework to create, configure, and use connectors. Certain Valtimo modules include their own connectors that become
available automatically for configuration when starting a Valtimo instance.

> **_NOTE:_** Connectors will be replaced by Plugins in the future

### Contract

Contract contains certain interfaces and events that Valtimo uses in several other modules. Consider it the 'glue'
between a module and Core.

Apart from this, contract sanitizes endpoint output when exceptions occur. A liquibase runner is available that allows
other modules to define liquibase configuration that is run on startup. Finally, Contract includes several helper

> classes, like the `QueryDialectHelper` bean that can be used in criteria queries to add JSON column clauses to the

Reviewer: ```suggestion
that can be used in criteria queries to add JSON column clauses to the
query.
```

that can be used in criteria queries to call DBMS specific functions in a generic way (e.g. finding JSON data).

query.

### Core

Core is the main module of Valtimo. It contains and builds upon the Camunda process engine that enables Valtimo to run
processes. Several endpoints and functions are available that provide task data and process data.

> It also includes choice fields. A generic way to create lists of values, that can be used by the application. Some

Reviewer 2: Maybe provide an example as to what choice fields can be used for?

Author: Not sure if choice fields are so important that we need to mention it at all actually. However, it was already there and I applied your suggestion.

security features used to control access to any Valtimo endpoints are also part of this module.

### Documents

> A document is the main data storage entity that can contain data for one or more processes. It can be linked to a case,

Reviewer: This is not completely correct. technically a document is the same as a case right now so it cannot be 'linked' to a case. 

```suggestion
A document is the main data storage entity that contains data of a case. Processes can act upon and change the data that is contained in the document. 
```

linear progress to the user without communication each specific step between milestones.

### Notes

Notes make it easier to collaborate with users on a case. It offers the ability to attach messages to a case that can be
read by other users.

### Plugins

Plugins are extensions on the Valtimo platform. These are often connections to external services, such as transactional
email and document generation. Plugins can be configured without writing code, and actions can be attached to BPMN
activities.

Some plugins are provided by the platform, but developers can create their own plugins as well. A more in-depth
description can be found [here](plugin-introduction.md).

### Process document

Process document facilitates the link between [documents](#documents) and BPMN processes. In order to use a process
with a document, the process needs to be linked to a document. Process document also provides access of document data
from inside a process.

### Resource

Resource is a module that describes how files can be uploaded and downloaded. This implemented by other modules,

> such as the [Temporary resource storage](#temporary-resource-storage) and [OpenZaak resource](#openzaak-resource).

Reviewer 2: I don't think the t in temporary needs to be capitalized.

It is used as a local test implementation that allows the application to start without having to call external services
when generating a document. This does *not* generate an actual document, but writes the calls to the application log
instead.

### Local mail

The local mail module is a stub implementation of the [mail module](#mail). It is used as a local test implementation
that allows the application to start without having to provide an actual implementation. This does *not* actually send
emails.

### Mail

Mail is the base module for mail implementations like [Local mail](#local-mail) and [Mandrill](#mandrill).

It orchestrates how Valtimo sends emails in a consistent way. Emails are filtered based on configured filters. For
example, emails to blacklisted addresses can be filtered out. Dispatching the email after filtering is facilitated by
another module.

### Mandrill

Mailchimp is a SaaS solution for sending emails. Mandrill is the name of
the [transactional mail addon](https://mailchimp.com/features/transactional-email/) that Mailchimp provides. The Valtimo

> Mandrill module provides a mail implementation of [Mail](#mail) that Valtimo uses to send emails..

Author: good catch.

> Mandrill module provides a mail implementation of [Mail](#mail) that Valtimo uses to send emails..

Author: ```suggestion
Mandrill module provides a mail implementation of [Mail](#mail) that Valtimo uses to send emails.
```
